federation/pkg/federation-controller/util/cluster_util.go

```diff
@@ -101,9 +105,16 @@ var KubeconfigGetterForSecret = func(secretName string) clientcmd.KubeconfigGett
 				return nil, fmt.Errorf("error in creating in-cluster client: %s", err)
 			}
 			data = []byte{}
-			secret, err := client.Secrets(namespace).Get(secretName)
+			var secret *api.Secret
+			for start := time.Now(); time.Since(start) < getSecretTimeout; time.Sleep(2 * time.Second) {
```

Comment:
nit: Please use standard retry periods from the utils package. Better still, use wait.Poll (or whatever it's called) from the utils package.